From 7225986962cdcce9413cff88d6f33c3549d92bc8 Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.king@canonical.com>
Date: Thu, 16 Sep 2021 13:29:44 +0000
Subject: [PATCH 2/3] stress-efivar: only exercise FS_IOC_*FLAGS if they are
 defined

C libraries such as musl don't yet define these flags, so only
exercise these ioctls if the flags are defined.

(backport of upstream stress-ng commit d7b4301d9a77e1b8cf0207ab1591ca5dac25d0f5)

Signed-off-by: Colin Ian King <colin.king@canonical.com>
---
 stress-efivar.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/stress-efivar.c b/stress-efivar.c
index def681e1..25551a85 100644
--- a/stress-efivar.c
+++ b/stress-efivar.c
@@ -108,8 +108,11 @@ static inline void efi_get_varname(char *dst, const size_t len, const efi_var *v
  */
 static int efi_get_variable(const args_t *args, const char *varname, efi_var *var)
 {
-	int fd, n, ret, rc = 0;
+#if defined(FS_IOC_GETFLAGS) &&		\
+    defined(FS_IOC_SETFLAGS)
 	int flags;
+#endif
+	int fd, n, ret, rc = 0;
 	char filename[PATH_MAX];
 	struct stat statbuf;
 
@@ -158,6 +161,8 @@ err_vars:
 		goto err_vars;
 	}
 
+#if defined(FS_IOC_GETFLAGS) &&		\
+    defined(FS_IOC_SETFLAGS)
 	ret = ioctl(fd, FS_IOC_GETFLAGS, &flags);
 	if (ret < 0) {
 		pr_err("%s: ioctl FS_IOC_GETFLAGS on %s failed, errno=%d (%s)\n",
@@ -173,6 +178,7 @@ err_vars:
 		rc = -1;
 		goto err_efi_vars;
 	}
+#endif
 
 err_efi_vars:
 	(void)close(fd);
-- 
2.32.0

